pandas/io/formats/csvs.py

```diff
@@ -127,58 +171,89 @@ def __init__(
         # and make sure sure cols is just a list of labels
         cols = self.obj.columns
         if isinstance(cols, ABCIndexClass):
-            cols = cols.to_native_types(
-                na_rep=na_rep,
-                float_format=float_format,
-                date_format=date_format,
-                quoting=self.quoting,
-            )
+            cols = self._convert_to_native_types(cols)
         else:
             cols = list(cols)
 
-        # save it
-        self.cols = cols
+        self._cols = cols
 
-        # preallocate data 2d list
-        ncols = self.obj.shape[-1]
-        self.data = [None] * ncols
+    def _convert_to_native_types(self, arg, **kwargs):
```

Comment:
I removed this method, but created a property ``_number_format`` (dict to store number formatting props). Now, instead of calling the method ``_convert_to_native_types`` I would call the original methods ``to_native_types`` with the ``_number_format`` passed alongside other kwargs if required.